File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        d={}
        
        count=0
        
        sum_=0
        
        for i in range(len(nums)):
            
            sum_+=nums[i]
            
            if sum_==k:
                count+=1
                
            if sum_-k in d:
                count+=d[sum_-k]
                
            if sum_ not in d:
                
                d[sum_]=1
                
            else:
                d[sum_]+=1
                
        return count
            

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        map<int,int> d;
        int sum=0;
        int count=0;
        
        for(int i=0;i<nums.size();i++){
            
            sum+=nums[i];
            
            if (sum==k){
                count++;
                
            }
            
             if (d.find(sum-k)!=d.end()){
                count+=d[sum-k];
            }
            
            if (d.find(sum)!=d.end()){
                d[sum]++;
            }
            else{
                d[sum]=1;
            }
            
        }
        
        return count;
        
    }
};